core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthChecker.java

```diff
@@ -105,8 +106,37 @@ static HealthChecker ofCpu(double targetSystemCpuUsage, double targetProcessCpuU
         return new CpuHealthChecker(targetSystemCpuUsage, targetProcessCpuUsage);
     }
 
+    /**
+     * Creates a new instance of {@link HealthChecker} which reports health based on
+     * cpu usage reported by {@link OperatingSystemMXBean}.
+     * Both system and process cpu usage must be (inclusive) lower than the specified threshold in order
+     * for the {@link HealthChecker} to report as healthy. If the {@link HealthChecker} is unable
+     * to find a suitable {@link OperatingSystemMXBean}, an exception is thrown on construction.
+     *
+     * @param targetSystemCpuUsage Target system CPU usage as a percentage (0 - 1).
+     * @param targetProcessCpuUsage Target process CPU usage as a percentage (0 - 1).
+     * @param degradedTargetSystemCpuUsage Degraded target system CPU usage as a percentage
+     *     (targetSystemCpuUsage - 1).
+     * @param degradedTargetProcessCpuLoad Degraded target process CPU usage as a percentage
+     *     (targetProcessCpuUsage - 1).
+     * @return an instance of {@link HealthChecker} configured with the provided CPU usage targets.
+     */
+    static HealthChecker ofCpu(double targetSystemCpuUsage, double targetProcessCpuUsage,
+                               double degradedTargetSystemCpuUsage, double degradedTargetProcessCpuLoad) {
+        return new CpuHealthChecker(targetSystemCpuUsage, targetProcessCpuUsage,
+                                    degradedTargetSystemCpuUsage, degradedTargetProcessCpuLoad);
+    }
+
     /**
      * Returns {@code true} if and only if the {@link Server} is healthy.
      */
     boolean isHealthy();
+
+    /**
+     * Returns the {@link HttpStatus} representing the health status of the {@link Server}.
+     * Override below method if you want more fine-grained health status.
+     */
+    default HealthStatus healthStatus() {
```

Comment:
Can you modify that `ScheduledHealthChecker` also uses the `healthStatus` API?

Reply:
Updated! 👍